CANTroller2/src/TemperatureSensorManager.h

```diff
@@ -0,0 +1,177 @@
+        sensor_location::ENGINE,
+        sensor_location::AMBIENT,
+        sensor_location::WHEEL_FL,
+        sensor_location::WHEEL_FR,
+        sensor_location::WHEEL_RL,
+        sensor_location::WHEEL_RR
+        };
+    std::map<sensor_location, DeviceAddress> known_addresses = {
+        {sensor_location::ENGINE, {0x28, 0x3c, 0xf3, 0xa7, 0x00, 0x00, 0x00, 0x00}},
+        {sensor_location::AMBIENT, {0x28, 0x14, 0xb3, 0xbf, 0x00, 0x00, 0x00, 0x00}}
+    };
+
+    std::map<sensor_location, TemperatureSensor> sensors;
+
+    // Assigns known addresses to Sensors. The sensors will have locations like ENGINE or AMBIENT
+    void assign_known_addresses() {
+    for (auto& known_address : known_addresses) {
+        // check to see if we have a known address that wasn't detected, print a warning if yes
+        auto detected_address_it = std::find_if(detected_addresses.begin(), detected_addresses.end(), [&](const DeviceAddress& detected_address) {
+            return std::equal(detected_address.begin(), detected_address.end(), known_address.second.begin());
+        });
+
+        if (detected_address_it != detected_addresses.end()) {
+            // The known address was detected, so assign it to the corresponding sensor
+            auto sensor_it = sensors.find(known_address.first);
+            if (sensor_it == sensors.end()) {
+                // The sensor doesn't exist yet, so create it and add it to the map
+                sensors.emplace(known_address.first, TemperatureSensor(known_address.first, *detected_address_it, &tempsensebus));
+            } else {
+                // The sensor already exists, so just update its address
+                sensor_it->second.set_address(*detected_address_it);
+            }
+        } else {
+            // The known address was not detected, so log a warning message
+            Serial.printf("Warning: Known sensor at location %s was not detected\n", TemperatureSensor::sensor_location_to_string(known_address.first).c_str());
+        }
+    }
+}
+
+    // Assign remaining addresses to any unassigned locations, in order of the sensor_locations enum
+    void assign_remaining_addresses() {
+        auto it = all_locations.begin();
+        for (auto& detected_address : detected_addresses) {
+            if (std::find_if(sensors.begin(), sensors.end(), [&](const std::pair<const sensor_location, TemperatureSensor>& pair) {
+                return std::equal(pair.second.get_address().begin(), pair.second.get_address().end(), detected_address.begin());
+            }) == sensors.end()) {
+                while (it != all_locations.end() && sensors.find(*it) != sensors.end()) {
+                    ++it;
+                }
+                if (it != all_locations.end()) {
+                    // The sensor doesn't exist yet, so create it and add it to the map
+                    sensors.emplace(*it, TemperatureSensor(*it, detected_address, &tempsensebus));
```

Comment:
maybe add a print statement here? could be useful for working out addresses when we make sensor changes

Reply:
Yeah I currently print out all of the sensors after initialization, could just print here and remove that method altogether, good call